Tasks/Maven/CodeAnalysis/mavencodeanalysis.ts

```diff
@@ -120,17 +86,64 @@ function findCandidateModules(directory:string):ModuleAnalysis[] {
         }
     });
 
+    // Special case: if the root turns up as a module, the automatic name won't do
+    result.forEach((module:ModuleAnalysis) => {
+        if (module.rootDirectory == tl.getVariable('build.sourcesDirectory')) {
+            module.moduleName = 'root';
+        }
+    });
+
     return result;
 }
 
-// Discover analysis results from enabled tools and associate them with the modules they came from
-function processAndAssignAnalysisResults(enabledCodeAnalysisTools:string[], modules:ModuleAnalysis[]):ModuleAnalysis[] {
+export function getCodeAnalysisStagingDirectory() {
+    var caStagingDir = path.join(tl.getVariable('build.artifactStagingDirectory'), ".codeAnalysis");
+    tl.mkdirP(caStagingDir);
+    return caStagingDir;
+}
+
+// Returns true if the given code analysis tool is enabled
+export function isCodeAnalysisToolEnabled(toolName:string): boolean {
+    // Get the list of enabled tools, return whether or not toolName is contained in it
+    var result = getEnabledCodeAnalysisTools().has(toolName);
+    if (result) {
+        // Looks like: 'SonarQube analysis is enabled.'
+        console.log(tl.loc('codeAnalysis_ToolIsEnabled', toolName));
```

Comment:
The caller of this function should be responsibe for logging. Otherwise, this function does too much. In general it's good to keep "query" like functions clean, because callers expect them not to have side effects.